app/models/task.rb

```diff
@@ -3,20 +3,32 @@
 # Table name: tasks
 #
 #  id          :bigint(8)        not null, primary key
-#  title       :string
 #  description :text
-#  section_id  :integer
+#  row_order   :integer
+#  title       :string
 #  created_at  :datetime         not null
 #  updated_at  :datetime         not null
+#  project_id  :bigint(8)
+#
+# Indexes
+#
+#  index_tasks_on_project_id  (project_id)
+#  index_tasks_on_row_order   (row_order)
+#
+# Foreign Keys
+#
+#  fk_rails_...  (project_id => projects.id)
 #
 
 class Task < ApplicationRecord
   include RankedModel
 
-  belongs_to :project
+  belongs_to :project, required: true
+
+  scope :row_order_asc, -> { order(row_order: :asc) }
 
   validates :title, length: { maximum: 250 }, presence: true
   validates :description, length: { maximum: 250 }, presence: true
 
-  ranks :row_order, with_same: :section_id
+  ranks :row_order, with_same: :project_id
```

Comment:
move before belongs_to 

rule is

...includes
....methods from 2rd party gems
....associations
..... scopes
...validations
....callbacks

read rails style guide